### ISSUE TITLE
`no-unused-vars` should not activate on object destructuring with `...rest`

### ISSUE THREAD

**Issue by judehunter** [eslint/eslint#13442]
I'm not sure if this has been mentioned before, I've only found issues about array - and not object - destructuring.
This issue might be controversial because I'm not exactly sure if what I'm proposing is a common practice, and/or if this change would make it harder to spot actual unused variables. It could be up to the user to turn it on or off

**What rule do you want to change?**
`no-unused-vars`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Special case for object destructuring when `...rest` is used.

**Please provide some example code that this change will affect:**

```js
// Basically, sometimes you want to get rid of an object field by destructuring
const {password, ...user} = req.user;
res.send(user); // The user object without the password is sent
```

**What does the rule currently do for this code?**
It marks `password` as an unused variable.

**What will the rule do after it's changed?**
Ignore it, similarly to arguments being ignored here (because the last one is used):
```js
function (a, b, c) {
  console.log(c);
}
```

**Are you willing to submit a pull request to implement this change?**
Not at this time.

**Comment by mdjermanovic** [comment]
Hi @judehunter, thanks for the issue!

I think this is the same as the existing [ignoreRestSiblings](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings) option:

```js
/* eslint no-unused-vars: ["error", { "ignoreRestSiblings": true }] */

const {password, ...user} = req.user; // ok
res.send(user);
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwiaWdub3JlUmVzdFNpYmxpbmdzXCI6IHRydWUgfV0gKi9cblxuY29uc3Qge3Bhc3N3b3JkLCAuLi51c2VyfSA9IHJlcS51c2VyOyAvLyBva1xucmVzLnNlbmQodXNlcik7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by judehunter** [comment]
@mdjermanovic my apologies, I must have skipped over that while reading the docs and was sure this behaviour did not exist.
